ui/components/ui/tabs/tab/tab.tsx

```diff
@@ -24,62 +21,42 @@ export const Tab = <TKey extends string = string>({
   // happy when being used in .tsx)
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   children,
-  textProps,
   disabled = false,
   ...props
 }: TabProps<TKey>) => {
-  const handleClick = (e: React.MouseEvent<HTMLDivElement>) => {
+  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
     e.preventDefault();
     if (!disabled && onClick) {
       onClick(tabIndex);
     }
   };
-
   return (
-    <Box
-      data-testid={dataTestId}
-      onClick={handleClick}
+    <Text
       key={tabKey}
-      role="tab"
-      tabIndex={isActive ? 0 : -1}
-      aria-selected={isActive}
-      aria-disabled={disabled}
-      {...props}
+      data-testid={dataTestId}
+      textAlign={TextAlign.Center}
+      color={TextColor.TextAlternative}
+      fontWeight={FontWeight.Medium}
+      variant={TextVariant.BodyMd}
       className={twMerge(
-        // Original class names for compatibility (e2e tests, etc.)
-        'tab',
-        isActive && 'tab--active',
-        disabled && 'tab--disabled',
-        isSingleTab && 'tab--single',
-        // Tailwind classes for actual styling
-        'border-b-2 border-muted text-alternative font-medium transition-colors duration-200 ease-out',
-        // Single tab variant (no border)
-        isSingleTab && 'text-default border-b-0',
-        // Active state
-        isActive &&
-          'text-default border-b-text-default transition-all duration-200 cubic-bezier(0.7, 0, 0.15, 1)',
-        // Disabled state
-        disabled && 'text-text-muted',
-        // Custom active class if provided and active
-        isActive && activeClassName,
+        'flex align-middle justify-center border-b-2 border-transparent px-0 py-1 transition-all duration-200 cubic-bezier(0.7, 0, 0.15, 1) hover:enabled:text-default',
+        isActive && 'text-default border-b-text-default',
+        disabled && 'text-muted',
         className,
       )}
+      {...props}
+      asChild
     >
-      <Text
-        textAlign={TextAlign.Center}
-        className={twMerge(
-          'block w-full min-w-[50px] bg-transparent font-medium transition-colors duration-[50ms] ease-out',
-          'disabled:text-inherit hover:enabled:text-default p-2',
-          buttonClassName,
-          textProps?.className,
-        )}
-        variant={TextVariant.BodyMd}
-        color={TextColor.Inherit}
-        {...textProps}
-        asChild
+      <button
+        role="tab"
+        tabIndex={isActive ? 0 : -1}
+        aria-selected={isActive}
+        aria-disabled={disabled}
+        onClick={handleClick}
+        disabled={disabled}
       >
-        <button disabled={disabled}>{name}</button>
-      </Text>
-    </Box>
+        {name}
+      </button>
+    </Text>
   );
```

Comment:
Updating `Tab` to but just a single button. Previously it was a `li` element with an `onClick` that wrapped a `button` element